lib/api/core/validation/types/enum.js

```diff
@@ -1,48 +1,68 @@
-
-/**
- * @constructor
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2017 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  */
-function EnumType () {
-  this.typeName = 'enum';
-  this.allowChildren = false;
-  this.allowedTypeOptions = ['values'];
-}
 
-util.inherits(EnumType, BaseConstructor);
+const BaseType = require('../baseType');
 
 /**
- * @param {TypeOptions} typeOptions
- * @param {*} fieldValue
- * @param {string[]} errorMessages
+ * @class EnumType
  */
-EnumType.prototype.validate = function enumTypeValidate (typeOptions, fieldValue, errorMessages) {
-  if (typeof fieldValue !== 'string') {
-    errorMessages.push('The field must be a string.');
-    return false;
+class EnumType extends BaseType {
+  constructor() {
+    super();
+    this.typeName = 'enum';
+    this.allowChildren = false;
+    this.allowedTypeOptions = ['values'];
   }
 
-  if (typeOptions.values.indexOf(fieldValue) === -1) {
-    errorMessages.push(`The field only accepts following values: "${typeOptions.values.join(', ')}".`);
-    return false;
-  }
+  /**
+   * @param {TypeOptions} typeOptions
+   * @param {*} fieldValue
+   * @param {string[]} errorMessages
+   */
+  validate(typeOptions, fieldValue, errorMessages) {
+    if (typeof fieldValue !== 'string') {
+      errorMessages.push('The field must be a string.');
+      return false;
+    }
 
-  return true;
-};
+    if (typeOptions.values.indexOf(fieldValue) === -1) {
+      errorMessages.push(`The field only accepts following values: "${typeOptions.values.join(', ')}".`);
+      return false;
+    }
 
-/**
- * @param {TypeOptions} typeOptions
- * @return {boolean|TypeOptions}
- * @throws InternalError
- */
-EnumType.prototype.validateFieldSpecification = function enumTypeValidateFieldSpecification (typeOptions) {
-  if (!typeOptions.hasOwnProperty('values') || !Array.isArray(typeOptions.values) || typeOptions.values.length === 0) {
-    return false;
+    return true;
   }
 
-  return !typeOptions.values.some(value => typeof value !== 'string');
-};
+  /**
+   * @param {TypeOptions} typeOptions
+   * @return {boolean|TypeOptions}
+   * @throws InternalError
+   */
+  validateFieldSpecification(typeOptions) {
+    if (!typeOptions.hasOwnProperty('values') || !Array.isArray(typeOptions.values) || typeOptions.values.length === 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 121 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)